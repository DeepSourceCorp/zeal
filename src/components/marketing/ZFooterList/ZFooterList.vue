<template>
  <ul :class="[SLOT_PADDING]">
    <li v-if="$slots.heading" class="leading-3 font-semibold pb-8 uppercase text-sm text-vanilla-400">
      <slot name="heading"></slot>
    </li>
    <span class="text-vanilla-300" :class="[ARRANGEMENT[arrange].classes]">
      <slot></slot>
    </span>
  </ul>
</template>

<script lang="ts">
import Vue from 'vue'
const SLOT_PADDING = 'px-6'
const ARRANGEMENT = {
  vertical: {
    text: 'vertical',
    classes: ''
  },
  horizontal: {
    text: 'horizontal',
    classes: 'flex flex-wrap'
  }
}
export default Vue.extend({
  name: 'ZFooterList',
  data() {
    return {
      ARRANGEMENT,
      SLOT_PADDING
    }
  },
  props: {
    arrange: {
      type: String,
      default: ARRANGEMENT.vertical.text,
      validator: (arrange) => Object.keys(ARRANGEMENT).includes(arrange)
    }
  }
})
</script>