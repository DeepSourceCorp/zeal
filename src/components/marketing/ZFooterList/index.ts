import ZFooterList from './ZFooterList.vue'

export default ZFooterList
