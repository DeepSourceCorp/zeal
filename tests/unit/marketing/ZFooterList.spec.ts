import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZFooterList from '../../../src/components/marketing/ZFooterList'

describe('ZFooterList component', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZFooterList, {
        localVue,
        ...options
      })
    }
  })
  it('renders a ZFooterList when the component is mounted', () => {
    expect(mountFunction().html()).toMatchSnapshot()
  })
  it('renders a ZFooterList with horizontal arrangement', () => {
    let mountedInstance = mountFunction({
      propsData: {
        arrange: 'horizontal'
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.html()).toContain('flex flex-wrap')
  })
  it('renders a ZFooterList when type is accordion', () => {
    let mountedInstance = mountFunction({
      propsData: {
        type: 'accordion'
      },
      slots: {
        heading: 'Dummy title',
        default: ['Something']
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.html()).toContain('chevron-right')
    expect(mountedInstance.text()).toContain('Something')
  })
  it('renders a ZFooterList when type is none', () => {
    let mountedInstance = mountFunction({
      propsData: {
        title: 'Dummy title'
      },
      slots: {
        default: ['Something']
      }
    })
    expect(mountedInstance.html()).toMatchSnapshot()
    expect(mountedInstance.html()).toContain('Dummy title')
    expect(mountedInstance.text()).toContain('Something')
  })
})
