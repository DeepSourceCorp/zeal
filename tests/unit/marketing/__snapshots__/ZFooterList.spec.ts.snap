// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZFooterList component renders a ZFooterList when the component is mounted 1`] = `
<ul class="px-6 md:px-4 lg:px-6">
  <!----> <span class="block"><!----> <span class="text-vanilla-300"></span></span>
</ul>
`;

exports[`ZFooterList component renders a ZFooterList when type is accordion 1`] = `
<ul class="px-6 md:px-4 lg:px-6"><span class="flex items-center cursor-pointer inline-block hover:opacity-75 select-none font-medium tracking-wider w-full leading-3 py-4 md:pt-0 uppercase text-sm text-vanilla-800"><span class="flex flex-1">Dummy title</span> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="stroke-1.5 transform z-icon w-6 h-6 z-icon--chevron-right text-vanilla-400">
    <polyline points="9 18 15 12 9 6"></polyline>
  </svg></span> <span class="block overflow-scroll transition-max-height duration-500 ease-in-out max-h-0"><span class="block text-sm md:text-md mx-1">Something</span></span>
  <!---->
</ul>
`;

exports[`ZFooterList component renders a ZFooterList when type is none 1`] = `
<ul class="px-6 md:px-4 lg:px-6" title="Dummy title">
  <!----> <span class="block"><!----> <span class="text-vanilla-300">Something</span></span>
</ul>
`;

exports[`ZFooterList component renders a ZFooterList with horizontal arrangement 1`] = `
<ul class="px-6 md:px-4 lg:px-6">
  <!----> <span class="block"><!----> <span class="text-vanilla-300 flex flex-wrap"></span></span>
</ul>
`;
